inference-engine/src/mkldnn_plugin/cpu_memory_desc_utils.cpp

```diff
@@ -0,0 +1,394 @@
+    for (size_t i = 0; i < outer_ndims; i++) {
+        outer_order[ie_order[i]] = i;
+    }
+    bool outer_is_correct_permutation_of_n =
+            std::find(outer_order.begin(), outer_order.end(), outer_ndims + 1) == outer_order.end();
+
+    if (!outer_is_correct_permutation_of_n)
+        IE_THROW() << "Unsupported case for conversion";
+
+    bool inner_block_are_dense = one_of(ie_strides.back(), 0, 1);  // stride 1 - is dense case, 0 - broad casted
+    for (int i = outer_ndims; i < ie_strides.size() - 1; i++) {
+        inner_block_are_dense &= (ie_strides[i] == ie_strides[i+1] * ie_blkdDims[i+1]);
+    }
+
+    if (!inner_block_are_dense)
+        IE_THROW() << "Unsupported case for conversion";
+
+    bool inner_pad_offsets_is_zero = std::all_of(ie_offsetsToData.begin() + outer_ndims, ie_offsetsToData.end(),
+                                                 [](size_t pad) { return  pad == 0; });
+
+    if (!inner_pad_offsets_is_zero)
+        IE_THROW() << "Unsupported case for conversion";
+
+    // Fill general memory desc fields
+    mkldnnDesc.data.format_kind = dnnl_blocked;
+    mkldnnDesc.data.data_type = memory::convert_to_c(MKLDNNMemory::convertToDataType(tDesc.getPrecision()));
+    mkldnnDesc.data.ndims = dims.size();
+    mkldnnDesc.data.offset0 = tDesc.getBlockingDesc().getOffsetPadding();
+    std::copy(dims.begin(), dims.end(), mkldnnDesc.data.dims);
+    std::copy(ie_offsetsToData.begin(), ie_offsetsToData.begin() + outer_ndims, mkldnnDesc.data.padded_offsets);
+    std::fill(mkldnnDesc.data.padded_dims, mkldnnDesc.data.padded_dims + outer_ndims, 1);
+    for (size_t i = 0; i < ie_order.size(); i++) {
+        auto idx = ie_order[i];
+        mkldnnDesc.data.padded_dims[idx] *= ie_blkdDims[i];
+    }
+
+    // Fill blocking desc
+    auto &dnn_blk_desc = mkldnnDesc.data.format_desc.blocking;
+    dnn_blk_desc.inner_nblks = inner_ndims;
+    std::copy(ie_blkdDims.end() - inner_ndims, ie_blkdDims.end(), dnn_blk_desc.inner_blks);
+    std::copy(ie_order.end() - inner_ndims, ie_order.end(), dnn_blk_desc.inner_idxs);
+    for (size_t i = 0; i < outer_ndims; i++) {
+        dnn_blk_desc.strides[i] = ie_strides[outer_order[i]];
+    }
+
+    return MKLDNNMemoryDesc(mkldnnDesc);
+}
+
+BlockedMemoryDesc MemoryDescUtils::convertToBlockedDescriptor(const MemoryDesc &desc) {
+    if (desc.getType() == MemoryDescType::Blocked) {
+        return *(desc.as<BlockedMemoryDesc>());
+    } else if (desc.getType() == MemoryDescType::Mkldnn) {
+        return MemoryDescUtils::convertToBlockedDescriptor(*(desc.as<MKLDNNMemoryDesc>()));
+    } else {
+        IE_THROW() << "Cannot convert to blocked memory descriptor. Unsupported memory desc type";
+    }
+}
+
+MemoryDescPtr MemoryDescUtils::applyUndefinedOffset(const MKLDNNMemoryDesc& desc) {
+    if (desc.getFormatKind() != dnnl_format_kind_t::dnnl_blocked)
+        return MKLDNNPlugin::make_unique<MKLDNNMemoryDesc>(desc);
```

Comment:
Yep, agree. I will add an exception.

Reply:
to avoid exception raising in conv was added additional checks:
https://github.com/maxnick/openvino/blob/b44707fc730f9892d732537c648819f827d4e2cd/inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.cpp#L397-L401
https://github.com/maxnick/openvino/blob/b44707fc730f9892d732537c648819f827d4e2cd/inference-engine/src/mkldnn_plugin/nodes/mkldnn_conv_node.cpp#L432-L436